### PR TITLE
fix a few typos

### PR DESCRIPTION
**Pull request by NickVolynkin**

### CODE REVIEW

File: blog/_posts/2016-02-17-stack-overflow-the-architecture-2016-edition.markdown [hunk 1]
```diff
@@ -70,7 +70,7 @@ Here are some rules that apply globally so I don't have to repeat them with ever
...
+First, you have to find us---that's [DNS](https://en.wikipedia.org/wiki/Domain_Name_System). Finding us needs to be fast, so we farm this out to [CloudFlare](https://www.cloudflare.com/) (currently) because they have DNS servers nearer to almost everyone around the world. We update our DNS records via an API and they do the "hosting" of DNS. But since we're jerks with deeply-rooted trust issues, we still have our own DNS servers as well. Should the apocalypse happen (probably caused by the GPL, [Punyon](https://twitter.com/JasonPunyon), or caching) and people still want to program to take their mind off of it, we'll flip them on.
```
Comment (NickVolynkin) [inline blog/_posts/2016-02-17-stack-overflow-the-architecture-2016-edition.markdown:73]: First,

File: blog/_posts/2016-05-03-stack-overflow-how-we-do-deployment-2016-edition.markdown [hunk 2]
```diff
@@ -333,7 +333,7 @@ Note: we do a chat notification for build as someone goes through the tiers. Her
...
+The database migrator we use is a very simple repo we could open source, but honestly, there are dozens out there and the "same migration against n databases" is fairly specific. The others are probably much better and ours is very specific to *only* our needs. The migrator connects to the Sites database, gets the list of databases to run against, and executes all migrations against every one (running multiple databases in parallel). This is done by looking at the passed-in migrations folder and loading it (once) as well as hashing the contents of every file. Each database has a `Migrations` table that keeps track of what has already been run. It looks like this (descending order):
```
Comment (NickVolynkin) [inline blog/_posts/2016-05-03-stack-overflow-how-we-do-deployment-2016-edition.markdown:336]: but honestly,

File: blog/_posts/2016-05-03-stack-overflow-how-we-do-deployment-2016-edition.markdown [hunk 3]
```diff
@@ -344,7 +344,7 @@ In memory, we compare the list of migrations that already ran to those needing t
...
+This will get its own post, but I wanted to hint at why we do all of this work at compile time. After all, I always advocate strongly for simplicity (yep, even in this 6,000-word blog post - the irony is not lost on me). You should only do something more complicated when you *need* to do something more complicated. This is one of those cases, for performance. [Samo](https://twitter.com/m0sa) does a lot of work to make our localizations have as little **runtime** impact as possible. We'll gladly trade a bit of build complexity to make that happen. While there are options such as [`.resx` files](https://msdn.microsoft.com/en-us/library/ekyft91f.aspx) or [the new localization in ASP.NET Core 1.0](https://github.com/aspnet/localization), most of these allocate more than necessary especially with tokenized strings. Here's what strings look like in our code:
```
Comment (NickVolynkin) [inline blog/_posts/2016-05-03-stack-overflow-how-we-do-deployment-2016-edition.markdown:347]: 6,000-word

**Comment by NickVolynkin** [comment]
Hi, @NickCraver!

I've fixed a few typos and some punctuation in the post. Looks like GitHub sometimes fails to show correct diffs for very long lines. For such cases I'll add a note, pointing at the change.

By the way, I highly recommend writing one sentence on a line. It makes diffs much more readable.

Oh, and thank you for the posts. I enjoyed reading them.

File: blog/_posts/2016-02-17-stack-overflow-the-architecture-2016-edition.markdown [hunk 4]
```diff
@@ -105,7 +105,7 @@ I'll go into why we're so overprovisioned in future posts, but the highlight ite
 
 ### Service Tier (IIS, ASP.Net MVC 5.2.3, .Net 4.6.1, and HTTP.SYS)
 
-Behind those web servers is the very similar "service tier." It's also running IIS 8.5 on Windows 2012R2. This tier runs internal services to support the production web tier and other internal systems. The two big players here are "Stack Server" which runs the tag engine and is based on `http.sys` (not behind IIS) and the Providence API (IIS-based). Fun fact: I have to set affinity on each of these 2 processes to land on separate sockets because Stack Server just steamrolls the L2 and L3 cache when refreshing question lists on a 2 minute interval.
+Behind those web servers is the very similar "service tier." It's also running IIS 8.5 on Windows 2012R2. This tier runs internal services to support the production web tier and other internal systems. The two big players here are "Stack Server" which runs the tag engine and is based on `http.sys` (not behind IIS) and the Providence API (IIS-based). Fun fact: I have to set affinity on each of these 2 processes to land on separate sockets because Stack Server just steamrolls the L2 and L3 cache when refreshing question lists on a 2-minute interval.
```
Comment (NickVolynkin) [inline blog/_posts/2016-02-17-stack-overflow-the-architecture-2016-edition.markdown:108]: 2-minute

File: blog/_posts/2016-02-17-stack-overflow-the-architecture-2016-edition.markdown [hunk 5]
```diff
@@ -115,7 +115,7 @@ We use [Redis](http://Redis.io/) for a few things here and it's rock solid. Desp
...
+We have an L1/L2 cache system with Redis. "L1" is HTTP Cache on the web servers or whatever application is in play. "L2" is falling back to Redis and fetching the value out. Our values are stored in the [Protobuf format](https://developers.google.com/protocol-buffers/), via [protobuf-dot-net](https://github.com/mgravell/protobuf-net) by Marc Gravell. For a client, we're using [StackExchange.Redis](https://github.com/StackExchange/StackExchange.Redis)---written in-house and open source. When one web server gets a cache miss in both L1 and L2, it fetches the value from source (a database query, API call, etc.) and puts the result in both local cache and Redis. The next server wanting the value may miss L1, but would find the value in L2/Redis, saving a database query or API call.
```
Comment (NickVolynkin) [inline blog/_posts/2016-02-17-stack-overflow-the-architecture-2016-edition.markdown:118]: an [el]1

File: blog/_posts/2016-05-03-stack-overflow-how-we-do-deployment-2016-edition.markdown [hunk 6]
```diff
@@ -43,7 +43,7 @@ I'm going ahead and inserting a set of section links here because this post got
...
+We deploy roughly 25 times per day to development (our CI build) just for Stack Overflow Q&A. Other projects also push many times. We deploy to production about 5-10 times on a typical day. A deploy from first push to full deploy is under 9 minutes (2:15 for dev, 2:40 for meta, and 3:20 for all sites). We have roughly 15 people pushing to the repository used in this post. The repo contains the code for these applications: [Stack Overflow](http://stackoverflow.com/) (every single Q&A site), [stackexchange.com](http://stackexchange.com/) (root domain only), [Stack Snippets](http://stacksnippets.net/) (for Stack Overflow JavaScript snippets), [Stack Auth](http://stackauth.com/) (for OAuth), [sstatic.net](http://sstatic.net/) (cookieless CDN domain), [Stack Exchange API v2](http://api.stackexchange.com/), [Stack Exchange Mobile](http://mobile.stackexchange.com/) (iOS and Android API), Stack Server (Tag Engine and Elasticsearch indexing Windows service), and Socket Server (our WebSocket Windows service).
```
Comment (NickVolynkin) [inline blog/_posts/2016-05-03-stack-overflow-how-we-do-deployment-2016-edition.markdown:46]: OAuth, it's a name.

File: blog/_posts/2016-05-03-stack-overflow-how-we-do-deployment-2016-edition.markdown [hunk 7]
```diff
@@ -72,7 +72,7 @@ Begin
 End
 {% endhighlight %}
 
-We've tested the migration works by running it against our local Q&A database of choice in SSMS and that the code on top of it works. Before deploying though, we need to make sure it runs *as a migration*. For example sometimes you may forget to put a [GO](https://msdn.microsoft.com/en-us/library/ms188037.aspx) separating something that must be the first or only operation in a batch such as creating a view. So, we test it in the runner. To do this, we run the `migrate.local.bat` you see in the screenshot above. The contents are simple:  
+We've tested the migration works by running it against our local Q&A database of choice in SSMS and that the code on top of it works. Before deploying though, we need to make sure it runs *as a migration*. For example, sometimes you may forget to put a [GO](https://msdn.microsoft.com/en-us/library/ms188037.aspx) separating something that must be the first or only operation in a batch such as creating a view. So, we test it in the runner. To do this, we run the `migrate.local.bat` you see in the screenshot above. The contents are simple:
```
Comment (NickVolynkin) [inline blog/_posts/2016-05-03-stack-overflow-how-we-do-deployment-2016-edition.markdown:75]: For example,

**Comment by NickCraver** [comment]
All good fixes - thank you!!

**Comment by NickVolynkin** [comment]
@NickCraver you're welcome! Cheers from the RUSO community. :)

**Pull request merged**